### ISSUE TITLE
SC2269 incorrectly says that an assignment does nothing

### ISSUE THREAD

**Issue by Brassrat** [koalaman/shellcheck#2722]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2269
- My shellcheck version (`shellcheck --version` or "online"): 0.9.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#! /usr/bin/env bash
xxx=$xxx
```

#### Here's what shellcheck currently says:
-- [SC2269](https://www.shellcheck.net/wiki/SC2269) (info): This variable is assigned to itself, so the assignment does nothing.

#### Here's what I wanted or expected to see:
A less definitive message. Something like:
This variable assignment will do nothing if the variable is already defined.

#### Here is why the message should be changed
If the script in question will be invoked when the variable already exists with some non-empty value then this assignment preserves the existing value; whereas a statement like 'xxx=' (to define xxx) would discard the existing value.
Isn't there a difference between a variable not being defined at all vs. having been defined with an empty value. So, having xxx=$xxx in a script may be intended to ensure that the variable is defined while preserving any value it may have.
I realize that shellcheck can't know if the variable will already be defined and having a message for this situation is probably useful.
I was just surprised to see how definitive the message is.

**Comment by pan64** [comment]
use:
aaa=${aaa:-}
